common/djangoapps/util/milestones_helpers.py

```diff
@@ -175,16 +174,20 @@ def get_required_content(course, user):
                 for content in milestone_path['content']:
                     required_content.append(content)
 
-    #local imports to avoid circular reference
+    # local imports to avoid circular reference
     from student.models import EntranceExamConfiguration
+    from courseware.access import has_access
     can_skip_entrance_exam = EntranceExamConfiguration.user_can_skip_entrance_exam(user, course.id)
-    # check if required_content has any entrance exam and user is allowed to skip it
+    # check if required_content has any entrance exam
+    # and user is allowed to skip it or user is member of staff
     # then remove it from required content
-    if required_content and getattr(course, 'entrance_exam_enabled', False) and can_skip_entrance_exam:
+    if required_content and getattr(course, 'entrance_exam_enabled', False) and \
+            (can_skip_entrance_exam or has_access(user, 'staff', course, course.id)):
```

Comment:
After looking at the changeset a bit further tonight, I have some thoughts:

1.  We seem to be blurring the context of `common\djangoapps\util\milestones_helpers.py` -- I think we need to move work out of this module which is not directly related to interactions with the edx-milestones service.  It does not all have to be part of this PR, but we can begin by moving the entrance exam logic from `get_required_content` to a new module in `lms\djangoapps\courseware`, perhaps named `entrance_exams.py`

2. Once we have the dedicated courseware module for entrance exams, hopefully the authorization approach will be more easy to envision.  Ideally we'll be able to simplify the workflow by exposing a straightforward operation to determine if the entrance exam can be skipped, which could incorporate both the `has_access` check as well as the call to EntranceExamConfiguration.

Thoughts?